core/templates/dev/head/components/forms/FormBuilder.js

```diff
@@ -935,7 +935,19 @@ oppia.directive('textAngularRte', [
           };
 
           $scope.stripFormatting = function(html) {
-            return $filter('sanitizeHtmlForRte')(html);
+            var whitelistedTags = [
+              'oppia-noninteractive-collapsible',
+              'oppia-noninteractive-image',
+              'oppia-noninteractive-link',
+              'oppia-noninteractive-math',
+              'oppia-noninteractive-tabs',
+              'oppia-noninteractive-video'
+            ];
+            // Strip out anything between and including <>,
+            // unless it contains the RTE tags.
+            var regex = new RegExp('(?!.*(' + whitelistedTags.join('|') +
```

Comment:
Pull this out into a filter, so that the only code in $scope.stripFormatting() is:

    return $filter('stripFormatting')(html, whitelistedTags);

You can put the new filter in filters.js. In addition, please add tests in filtersSpec.js that exercise this filter.